core/src/main/java/org/kiwix/kiwixmobile/core/main/TabsAdapter.kt

```diff
@@ -0,0 +1,170 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.main
+
+import android.annotation.SuppressLint
+import android.text.TextUtils
+import android.view.View
+import android.view.ViewGroup
+import android.widget.FrameLayout
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.constraintlayout.widget.ConstraintSet
+import androidx.core.content.ContextCompat
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.card.MaterialCardView
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.extensions.getAttribute
+import org.kiwix.kiwixmobile.core.extensions.tint
+import org.kiwix.kiwixmobile.core.utils.DimenUtils.getToolbarHeight
+import org.kiwix.kiwixmobile.core.utils.DimenUtils.getWindowHeight
+import org.kiwix.kiwixmobile.core.utils.DimenUtils.getWindowWidth
+import org.kiwix.kiwixmobile.core.utils.ImageUtils.getBitmapFromView
+import org.kiwix.kiwixmobile.core.utils.StyleUtils.fromHtml
+
+class TabsAdapter internal constructor(
+  private val activity: AppCompatActivity,
+  private val webViews: List<KiwixWebView>,
+  private val painter: NightModeViewPainter
+) : RecyclerView.Adapter<TabsAdapter.ViewHolder>() {
+  private var listener: TabClickListener? = null
+  var selected = 0
+  @SuppressLint("ResourceType") override fun onCreateViewHolder(
+    parent: ViewGroup,
+    viewType: Int
+  ): ViewHolder {
+    val context = parent.context
+    val margin16 = context.resources.getDimensionPixelSize(R.dimen.activity_horizontal_margin)
+    val contentImage = ImageView(context)
```

Comment:
use `apply` when you are configuring objects. It should make this code a lot prettier as the nodes will be more structurally visible

Reply:
https://github.com/kiwix/kiwix-android/pull/2452/commits/ac979a2534bac443b4fb2dc2f10a4542d994f5b9